backend/RayTracing/Classes/Models/Object3D.py

```diff
@@ -16,10 +16,10 @@ def __init__(self, center=Vector(0, 0, 0), color=Color(), specular=100, reflecti
         self.specular = specular
         self.reflection = reflection
 
-    def intersection(self, ray):
+    def intersection(self, ray, tMin, tMax):
         pass
```

Comment:
Why do we need tMin and tMax as a method parameter?

Reply:
So tMin and tMax specify the range in which an intersection can be found. We compare if t (the distance of the point found) is between tMin and tMax. This is useful, in order to avoid situations where a sphere in the back casts a shadow in a sphere that's in front or that a sphere behind the camera is displayed.